smartsim/_core/generation/generator.py

```diff
@@ -24,219 +24,224 @@
-    ) -> None:
-        """Initialize a generator object
-
-        if overwrite is true, replace any existing
-        configured applications within an ensemble if there
-        is a name collision. Also replace any and all directories
-        for the experiment with fresh copies. Otherwise, if overwrite
-        is false, raises EntityExistsError when there is a name
-        collision between entities.
-
-        :param gen_path: Path in which files need to be generated
-        :param overwrite: toggle entity replacement
-        :param verbose: Whether generation information should be logged to std out
-        """
-        self._writer = ApplicationWriter()
-        self.gen_path = gen_path
-        self.overwrite = overwrite
-        self.log_level = DEBUG if not verbose else INFO
+    def __init__(self, root: pathlib.Path) -> None:
+        """Initialize a Generator object
 
-    @property
-    def log_file(self) -> str:
-        """Returns the location of the file
-        summarizing the parameters used for the last generation
-        of all generated entities.
-
-        :returns: path to file with parameter settings
+        The class handles symlinking, copying, and configuration of files
+        associated with a Jobs entity. Additionally, it writes entity parameters
+        used for the specific run into the "smartsim_params.txt" settings file within
+        the Jobs log folder.
         """
-        return join(self.gen_path, "smartsim_params.txt")
+        self.root = root
+        """The root path under which to generate files"""
 
-    def generate_experiment(self, *args: t.Any) -> None:
-        """Run ensemble and experiment file structure generation
+    def _generate_job_root(self, job: Job, job_index: int) -> pathlib.Path:
+        """Generates the root directory for a specific job instance.
 
-        Generate the file structure for a SmartSim experiment. This
-        includes the writing and configuring of input files for a
-        application.
+        :param job: The Job instance for which the root directory is generated.
+        :param job_index: The index of the Job instance (used for naming).
+        :returns: The path to the root directory for the Job instance.
+        """
+        job_type = f"{job.__class__.__name__.lower()}s"
+        job_path = self.root / f"{job_type}/{job.name}-{job_index}"
+        return pathlib.Path(job_path)
 
-        To have files or directories present in the created entity
-        directories, such as datasets or input files, call
-        ``entity.attach_generator_files`` prior to generation. See
-        ``entity.attach_generator_files`` for more information on
-        what types of files can be included.
+    def _generate_run_path(self, job: Job, job_index: int) -> pathlib.Path:
+        """Generates the path for the \"run\" directory within the root directory
+        of a specific Job instance.
```

Comment:
Minor nit: In theory, there is no need to escape the `"`s here!
```suggestion
        """Generates the path for the "run" directory within the root directory
        of a specific Job instance.
```